airbyte-integrations/connectors/source-trello/source_trello/source.py

```diff
@@ -0,0 +1,227 @@
+
+
+class Boards(TrelloStream):
+    """Return list of all boards.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-attached-segments-1
+    Endpoint: https://api.trello.com/1/members/me/boards
+    """
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "members/me/boards"
+
+
+class Cards(ChildStreamMixin, TrelloStream):
+    """Return list of all cards of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-cards-get
+    Endpoint: https://api.trello.com/1/boards/<id>/cards/all
+    """
+
+    parent_stream_class = Boards
+    limit = 20000
+    extra_params = {"customFieldItems": "true"}
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/cards/all"
+
+
+class Checklists(ChildStreamMixin, TrelloStream):
+    """Return list of all checklists of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-checklists-get
+    Endpoint: https://api.trello.com/1/boards/<id>/checklists
+    """
+
+    parent_stream_class = Boards
+    extra_params = {"fields": "all", "checkItem_fields": "all"}
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/checklists"
+
+
+class Lists(ChildStreamMixin, TrelloStream):
+    """Return list of all lists of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-lists-get
+    Endpoint: https://api.trello.com/1/boards/<id>/lists
+    """
+
+    parent_stream_class = Boards
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/lists"
+
+
+class Users(ChildStreamMixin, TrelloStream):
+    """Return list of all members of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-members-get
+    Endpoint: https://api.trello.com/1/boards/<id>/members
+    """
+
+    parent_stream_class = Boards
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/members"
+
+
+class Actions(ChildStreamMixin, IncrementalTrelloStream):
+    """Return list of all actions of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-boardid-actions-get
+    Endpoint: https://api.trello.com/1/boards/<id>/actions
+    """
+
+    parent_stream_class = Boards
+    limit = 1000
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/actions"
+
+
+class TrelloAuthenticator(HttpAuthenticator):
+    def __init__(
+        self,
+        token: str,
+        key: str,
```

Comment:
Reuse the CDK class:
```
class TokenAuthenticator(HttpAuthenticator):
    def __init__(self, token: str, auth_method: str = "Bearer", auth_header: str = "Authorization"):
```

Reply:
No need. HttpAuthenticator is also a CDK class.

Reply:
the TokenAuthenticator class already provides a part of your attributes

Reply:
But also provide unnecessary attributes. So, I didn't see benefits for reuse TokenAuthenticator instead of HttpAuthenticator.